### PR TITLE
Hipify caffe2/utils/math_gpu.cu

### PR DESCRIPTION
**Pull request by rohithkrn**
This PR adds caffe2/utils/math_gpu.cu to pyHipify

@bddppq @petrex

### CODE REVIEW

**Comment by bddppq** [comment]
Thanks for working removing the duplication in math_hip!

File: tools/amd_build/pyHIPIFY/hipify_python.py [hunk 1]
```diff
@@ -1349,11 +1350,15 @@ def hipify(
...
         for filepath in all_files:
-            get_kernel_template_params(
-                filepath,
-                KernelTemplateParams,
-                CAFFE2_TEMPLATE_MAP if hipify_caffe2 else PYTORCH_TEMPLATE_MAP)
+            if "math_gpu.cu" not in os.path.basename(filepath):
+                get_kernel_template_params(
+                    filepath,
+                    KernelTemplateParams,
+                    CAFFE2_TEMPLATE_MAP if args.hipify_caffe2 else PYTORCH_TEMPLATE_MAP)
```
Comment (bddppq) [inline tools/amd_build/pyHIPIFY/hipify_python.py]: remove `args.`

File: tools/amd_build/pyHIPIFY/hipify_python.py [hunk 2]
```diff
@@ -1349,11 +1350,15 @@ def hipify(
...
         KernelTemplateParams = {}
+        # ignore caffe2/utils/math_gpu.cu for static_casts as PowKernel
+        # conflicts with kernel in caffe2/operators/pow_op.cu.
+        # Need to figure out a long-term solution.
         for filepath in all_files:
-            get_kernel_template_params(
-                filepath,
-                KernelTemplateParams,
-                CAFFE2_TEMPLATE_MAP if hipify_caffe2 else PYTORCH_TEMPLATE_MAP)
+            if "math_gpu.cu" not in os.path.basename(filepath):
```
[the remark this answers is not in the document]
Comment (rohithkrn) [reply tools/amd_build/pyHIPIFY/hipify_python.py]: Any suggestions? I am thinking `PowOpKernel`

File: tools/amd_build/build_caffe2_amd.py [hunk 3]
```diff
@@ -22,6 +22,7 @@
     "caffe2/**/*_test*",
     "caffe2/core/*",
     "caffe2/db/*",
+    "caffe2/utils/math_gpu*",
```
[the remark this answers is not in the document]
Comment (rohithkrn) [reply tools/amd_build/build_caffe2_amd.py]: Other files like `Gpu*` requires more changes. @ashishfarmer is going to include them in the PR enabling `top_k` operator.

File: tools/amd_build/pyHIPIFY/hipify_python.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bddppq) [reply tools/amd_build/pyHIPIFY/hipify_python.py]: Yep sgtm (but please add a comment there so people don't change it back)

File: tools/amd_build/build_caffe2_amd.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bddppq) [reply tools/amd_build/build_caffe2_amd.py]: `Gpu*` should all be .cuh files? So it doesn't hurt to hipify them I think (they will just get ignored and not compiled)

**Comment by bddppq** [comment]
@pytorcbot test this please

**Comment by bddppq** [comment]
@BIT-silence Kindly ping for review :-)

**Comment by bddppq** [comment]
pr/py2-clang7-rocmdeb-ubuntu16.04 is known broken in master, fix here in #13662

**Review by bddppq — APPROVED** [review]

**Pull request closed without merging**